### DISCUSSION TITLE
Name of this package

### DISCUSSION THREAD

**Discussion by sibirrer** [LSST-strong-lensing/slsim#38] [Q&A] [answered]
The name **sim-pipeline** (and the currently used `import sim_pipeline` statements) are not uniquely recognizable for this package, and we would like to come up with a better name.

Please add your suggestions to this discussion and/or vote on the suggestions using the up arrow below each suggestion. The voting will close on **Sep 24, 2023**.

**Comment by ajshajib** [comment]
**lenSim**

**Reply by nkhadka21** [reply]
Or lenSimPipe?

**Reply by ajshajib** [reply]
Creating a separate thread for @nkhadka21's suggestion so that people have the option to press the up arrow to vote on this.

**Comment by ajshajib** [comment]
**cosmoLens**

**Reply by stefanschuldt** [reply]
In the first moment I was thinking that's the [ERC funded Collaboration name](https://cosmiclens.epfl.ch) from Fred Courbin and Dominique Sluse. Not exactly the same name but very close, so I would go for something else to avoid confusion although it sounds good.

**Reply by martin-millon** [reply]
Yes, the name of the grant is cosmiclens.

**Comment by ajshajib** [comment]
For a name with a pun, **cosMonocle**. 🧐

**Comment by ajshajib** [comment]
**lenSimPipe**, suggested by @nkhadka21.

**Comment by drphilmarshall** [comment]
How about `lsst-sl-sims` as in:

`from lsst-sl-sims import GalaxyGalaxyLensPop`

I think PEP8 style is to choose a short, lowercase (not camelCase or TitleCase) word for a package name, so might be better to have something without any hyphens. Seems important to list LSST in the package name though, unless we aspire to providing more general image simulation tools...

**Reply by ajshajib** [reply]
I think the aspiration is, indeed, to provide a more general tool. 🙂 Yes, the regular name can have uppercase letters to clarify the wording/abbreviations, but the import statement within Python should have all lowercase. For example, if **lenSim** is the chosen name,  then that's how we would refer to it in a paper or in writing, but then the import statement would look like `import lensim`.

**Reply by ajshajib** [reply]
I think, one proposal based on your suggestion can be **SLSims**, with the import statement being `import slsims`.

**Comment by ajshajib** [comment]
**SLPipe**

**Reply by ajshajib** [reply]
Con: this name sounds like data analysis would also fall within the scope of the software, not just simulations.

**Comment by ajshajib** [comment]
**lensPipe**

**Comment by Nikki1510** [comment]
SimuLens

**Comment by Nikki1510** [comment]
Lensify

**Comment by drphilmarshall** [comment]
I asked ChatGPT, see below for its best suggestions and the task I gave it, for those who are interested. It too came up with `lensSim` and `SimuLens`, interestingly - and also seems to prefer CamelCase for package names. From this list I think **LensForge** might be the best. 

`from lensforge import GalaxyGalaxyLensPop`

"Suggest 100 possible names for a python software package that provides the simulation tools for the purpose of conducting strong gravitational lensing science in wide-field optical imaging surveys such as Rubin Observatory's LSST and the Roman Space Telescope surveys. All the code is open-source, modular, object-oriented and collaboratively developed: it supports the simulation of populations of various kinds of gravitational lens, including lensed galaxies (Einstein Rings), lensed AGN, and lensed supernovae, and the realization of mock cutout images of those systems in the LSST data. In future, modules will be added to support simulation of images from Roman and other telescopes. Display the names in a concise table."

Certainly! Here are 100 possible names for your Python software package for strong gravitational lensing simulations:

| Name Suggestions                |
|---------------------------------|
| LensCraft                        |
| LensSim                          |
| LensVista                        |
| AstroFocal                       |
| Gravimager                       |
| LensGenerator                    |
| Lensiverse                       |
| LensCraftPro                     |
| LensBuilder                      |
| LensForge                        |
| LensPro                          |
| LensMagic                        |
| LensMaker                        |

**Comment by drphilmarshall** [comment]
**SLSpy**, for "strong lens simulations in python"

`from slspy import GalaxyGalaxyLensPop`

**Comment by drphilmarshall** [comment]
I saw LensFactory suggested somewhere and thought that was pretty good:

`from lensfactory import GalaxyGalaxyLensPop`

**Reply by ajshajib** [reply]
Both **lensFactory** and **lensForge** were in the bunch of proposed names, but they received zero votes until last Sunday, so I deleted them to decrease clutter, but good to have your vote on them now. 🙂

**Comment by ajshajib** [comment, marked as answer]
**SLSim** (based on @drphilmarshall's suggestion)

`from slsim import GalaxyGalaxyLensPop`
